src/Pact/Native.hs

```diff
@@ -153,6 +156,19 @@ enforceDef = defNative "enforce" enforce
     enforce' i as = argsError i as
     {-# INLINE enforce' #-}
 
+    enforceLazy i [cond, msg] = do
+      cond' <- reduce cond
+      case cond' of
+        TLiteral (LBool b') _ ->
+          if b' then
+            return (TLiteral (LBool True) def)
+          else reduce msg >>= \case
+            TLitString msg' -> failTx (_faInfo i) $ pretty msg'
+            e -> evalError' i $ "Invalid message argument, expected string " <> pretty e
+        _ -> reduce msg >>= argsError i . reverse . (:[cond'])
+    enforceLazy i as = mapM reduce as >>= argsError i
+    {-# INLINE enforceLazy #-}
```

Comment:
Does this really need to be `INLINE`? Is it called that often or in tight loops? I would prefer `INLINABLE` and let GHC decide.

Reply:
Tbh ghc probably already inlines it, there's only 1 callsite. I suspect the pragma is useless, it exists since I kinda copy pasted a block.